File: be-service/app/routers/restock_recommendation.py
from fastapi import APIRouter

from app.models.restock_recommendation import (
    RestockRecommendationRequest
)

from app.controllers.restock_recommendation_controller import (
    generate
)

router = APIRouter(
    prefix="/restock-recommendation",
    tags=["restock-recommendation"]
)

@router.post("")
def get_recommendation(
    payload:
    RestockRecommendationRequest
):

    result = generate(
        payload.budget,
        payload.horizon_days
    )

    return {
        "success": True,
        "message":
            "Recommendation generated successfully",
        "data":
            result
    }

File: fe-service/src/lib/restockRecommendation.ts
import api from "./api";

export async function generateRestockRecommendation(
  budget: number,
  horizon_days: number
) {
  const response = await api.post(
    "/api/v1/restock-recommendation",
    {
      budget,
      horizon_days,
    }
  );

  return response.data.data;
}